### PR TITLE
refactor(auction): settle detect excessive want before atomicRearrange

### LINKED ISSUE #7282: Auction should pay attention to `want` in bid offers
## What is the Problem Being Solved?

The auction currently ignores `want`s specified in orders,  relying on Zoe's guarantees to ensure they're met. 

## Description of the Design

The auction should explicitly notice the want, and if a reallocation would only partially satisfy the offer, it should exit the seat instead.

## Security Considerations

minor.

## Scaling Considerations

None.

## Test Plan

verify that the following offer is handled even if the offer with a `want` might have exhausted the goods for sale.

### PR DESCRIPTION
**Pull request by Chris-Hibbert**
closes: #7282

## Description

The auction explicitly notices the proposal's want, and if a reallocation would only partially satisfy the offer, it exits the seat instead.

### Security Considerations

cleanly exits a position, rather than leaving it for Zoe to clean up.

### Scaling Considerations

Inconsequential.

### Documentation Considerations

None

### Testing Considerations

There's a new test/

### CODE REVIEW

File: packages/inter-protocol/test/auction/test-auctionContract.js [hunk 1]
```diff
@@ -400,6 +411,33 @@ test.serial('discount bid settled', async t => {
...
+
+  const seat = await driver.bidForCollateralSeat(
+    currency.make(250n),
+    collateral.make(5000n),
+    makeRatioFromAmounts(currency.make(120n), currency.make(100n)),
+    undefined,
+    collateral.make(5000n),
+  );
+  await t.throwsAsync(() => E(seat).getOfferResult(), {
+    message: 'seat has been exited',
```
Comment (turadg) [inline packages/inter-protocol/test/auction/test-auctionContract.js:435]: how do we test the exit reason, 'unable to satisfy want'?

**Review by turadg — COMMENTED** [review]
Not sure about test coverage yet.

Also, a `refactor:` doesn't change the behavior of the code. This is a `feat`(ure)

File: packages/inter-protocol/test/auction/test-auctionContract.js (hunk 1 above)
Comment (Chris-Hibbert) [reply packages/inter-protocol/test/auction/test-auctionContract.js:435]: https://github.com/Agoric/documentation/pull/791

added to the test

**Comment by Chris-Hibbert** [comment]
>  a refactor: doesn't change the behavior of the code. This is a feat(ure)

Fixed.

**Review by turadg — APPROVED** [review]

File: packages/zoe/src/zoeService/types.js [hunk 2]
```diff
@@ -214,7 +214,7 @@
  * @property {() => Promise<Allocation>} getFinalAllocation
  * return a promise for the final allocation. The promise will resolve after the
  * seat has exited.
- * @property {() => Subscriber<AmountKeywordRecord>} getExitSubscriber returns a subscriber that
+ * @property {() => Subscriber<any>} getExitSubscriber returns a subscriber that
```
Comment (turadg) [inline packages/zoe/src/zoeService/types.js]: please no `any`. it's it `void`?

Comment (Chris-Hibbert) [reply packages/zoe/src/zoeService/types.js]: According to `contractFacet/types.js`, 
```
/**
 * @typedef {any} Completion
 * Any passable non-thenable. Often an explanatory string.
 */
```
I'll switch to `Completion`.

**Review by turadg — APPROVED** [review]

**Pull request merged**